sql/core/src/test/scala/org/apache/spark/sql/connector/DataSourceV2SQLSuite.scala

```diff
@@ -3261,3 +3278,7 @@ class FakeV2Provider extends SimpleTableProvider {
     throw new UnsupportedOperationException("Unnecessary for DDL tests")
   }
 }
+
+class TestCreateTableReserveSchemaNullabilityCatalog extends InMemoryCatalog {
```

Comment:
How about `ForbidSchemaNullabilityCatalog` ?

Reply:
"forbid" seems not accurate here, would prefer to use "reserve" to represent respecting the query schema nullability

Reply:
I means the name seems too long. Could you rename it with simple one ?

Reply:
Maybe "ReserveSchemaNullabilityCatalog"?